### ISSUE TITLE
feat(cli): restore background removal for input images

### ISSUE THREAD

**Issue by rasagyavatsal** [rasagyavatsal/CropCheckUp#121]
## Problem

The current Python CLI classifies raw input photos without removing their backgrounds. The user remembers removing backgrounds from images when preparing the training dataset, but no longer remembers which model or notebook performed that operation. New camera/gallery photos can therefore reach the classifier with backgrounds unlike the prepared training images.

This issue concerns **new images passed to `cropcheckup.py`**, not reprocessing the dataset or retraining the classifier.

Two related preprocessing problems exist in `prepare_image()`:

- It discards alpha and only makes pixels with `alpha == 0` black. Partially transparent pixels become fully opaque: a red pixel with alpha `1/255` becomes `[255, 0, 0]`. Background-removal masks commonly have partially transparent edges, so those edges need proper compositing.
- It center-crops with `ImageOps.fit()`, whereas `train.py` uses `image_dataset_from_directory()` without cropping or padding. The training loader resizes the full image. For rectangular inputs, the CLI can discard leaf edges or disease markings and uses different geometry from training.

Current code locations: `cropcheckup.py` (`build_parser`, `prepare_image`, `classify`, `main`), `train.py` (training/validation dataset loading), `requirements.txt`, and `tests/test_cropcheckup.py`.

## Evidence and recoverable assets

The older app already removed backgrounds before classification. Commit `c055db75fef754efc081dfb98987101a39fb6b8d` introduced a Flutter `BackgroundRemovalService` using `image_background_remover`, with transparent backgrounds mapped to black for inference.

The subsequent browser implementation contains a recoverable ONNX model and its processing code at commit **`b8036567f6af4208d61202167a3022549ebfdc75`**:

- [Background-removal model](https://github.com/rasagyavatsal/CropCheckUp/blob/b8036567f6af4208d61202167a3022549ebfdc75/landing/public/models/background_removal.onnx): `landing/public/models/background_removal.onnx` (4,574,861 bytes; Git blob `4f21f9afd34a64b83aea71744ce5bc013eba50fc`).
- [Model license](https://github.com/rasagyavatsal/CropCheckUp/blob/b8036567f6af4208d61202167a3022549ebfdc75/landing/public/models/BACKGROUND_REMOVER_LICENSE.txt): BSD-3-Clause, copyright 2025 Netesh Paudel. Preserve the full notice.
- [Inference and mask processing](https://github.com/rasagyavatsal/CropCheckUp/blob/b8036567f6af4208d61202167a3022549ebfdc75/landing/src/services/backgroundRemoval.ts): `landing/src/services/backgroundRemoval.ts`.
- [Image-processing constants](https://github.com/rasagyavatsal/CropCheckUp/blob/b8036567f6af4208d61202167a3022549ebfdc75/landing/src/services/imageProcessing.ts): `landing/src/services/imageProcessing.ts`.

Use `git show <commit>:<path>` to inspect/recover these files without checking out the old application. The old `backgroundRemoval.ts` uses:

- Input name `input.1`, float32 NCHW shape `(1, 3, 320, 320)`.
- RGB channels normalized as `(channel / 255 - mean) / std`, with mean `[0.485, 0.456, 0.406]` and std `[0.229, 0.224, 0.225]`.
- The first output as a mask, resized to the source image dimensions.
- Edge enhancement and smoothing, followed by alpha thresholds: below `0.45` becomes transparent; above `0.55` becomes opaque; intermediate values are linearly mapped to alpha. Read the full source when porting these operations; these notes do not replace it.

Verify the actual ONNX input/output metadata when implementing. The old browser flow capped the image's longest dimension at 1024; decide and document whether to retain this working-size limit. Its center-crop helper should not be copied into the new classifier preprocessing because it conflicts with the current training resize behavior.

**Provenance limitation:** this is the model previously used for app inputs. We have not established that it is the exact model used to prepare the dataset, and should not describe it as such. The public [Kaggle training notebook](https://www.kaggle.com/code/rasagyavatsal/cropcheckup) loads the already-prepared dataset and contains no background-removal step. The [dataset description](https://www.kaggle.com/datasets/rasagyavatsal/cropcheckup-dataset) does not identify that model either. Those sources and the repository history were checked; the original model remains unidentified.

## Proposed fix

1. Restore the historical ONNX asset to `models/background_removal.onnx` with its license. Update model attribution/license documentation and add a compatible Python ONNX Runtime dependency for the documented Python 3.12 environment.
2. Run background removal locally by default for CLI input images. Add a focused Python preprocessing helper/module as appropriate. Reuse the recovered model's input normalization and mask processing; initialize its inference session only when needed and reuse it across repeated calls.
3. Apply this pipeline:

   `Decode → correct EXIF orientation → background removal → composite onto black → resize full image to 224×224 → float32 RGB batch → Keras classification`.

   Preserve source transparency when applying the new mask. Composite partial alpha correctly rather than retaining full RGB values whenever alpha is nonzero. Resize using bilinear interpolation without a center crop, matching the training pipeline's geometry.
4. Add `--save-processed PATH` to save a lossless PNG of the **final 224×224 RGB image actually fed to the classifier**, for visual inspection. Preserve the original source image, including rejecting an output path that resolves to the input file.
5. Load the ONNX asset from the selected `--model-dir`, alongside the classifier assets, and document the added resource requirement. Provide actionable errors for missing dependencies/assets, invalid model output, or inference/save failures; do not silently classify the original photo after segmentation fails.
6. Keep the existing prediction fields, `--top-k`, and text/JSON output behavior. Send diagnostics to stderr so `--json` remains parseable. Thread the new controls through direct Python calls without unnecessarily breaking existing positional arguments.
7. Update the README with dependency/setup instructions, model provenance, default behavior, flags, and examples:

   ```sh
   python cropcheckup.py leaf.jpg
   python cropcheckup.py leaf.jpg --save-processed processed.png
   python cropcheckup.py leaf.jpg --json
   ```

The Keras classifier already contains a rescaling layer and a final softmax. Continue feeding float32 RGB values in `[0, 255]`, shape `(1, 224, 224, 3)`; do not add another classifier normalization or softmax. The ONNX normalization above applies only to the background-removal model.

## Validation and acceptance criteria

- [ ] A normal CLI invocation runs the recovered background-removal model before disease classification; processing stays local and needs no network at inference time.
- [ ] Fully transparent, partially transparent, and opaque inputs composite correctly on black. For example, 50%-transparent red becomes approximately `[128, 0, 0]`, not `[255, 0, 0]`.
- [ ] EXIF orientation and rectangular images are handled correctly; full-image resizing preserves content that the previous center crop removed.
- [ ] `--save-processed` produces a PNG whose pixels match the actual classifier input, and never overwrites the source image.
- [ ] Missing/corrupt segmentation resources, invalid masks, and output-save errors produce clear failures with nonzero exit status. JSON stdout remains valid on success.
- [ ] Add meaningful deterministic tests using controlled masks/stubs for preprocessing and CLI behavior; include a real-model integration smoke test for the restored ONNX asset and bundled Keras classifier. Update existing tests for the new default behavior.
- [ ] Run `python -m unittest discover -s tests -p 'test_*.py' -v` in the documented Python 3.12 environment. Before this change, all four existing tests passed; they did not cover segmentation, partial transparency, or training/inference resize consistency.
- [ ] Visually inspect representative real leaf photos and saved outputs, especially diseased regions and leaf edges. Record the examples and any failure modes. Do not claim an accuracy improvement based on confidence scores or synthetic-image smoke tests; accuracy comparison requires labeled examples.
- [ ] README, requirements, model license, and attribution reflect the shipped behavior and assets.

## Scope and handoff notes

- Leave the training dataset and classifier weights unchanged. No retraining is required for this implementation task.
- The original dataset-preparation model is unknown; implement using the recoverable app model and state that limitation accurately.
- Separate review observation, **not part of this issue**: direct `classify(..., top_k=0)` calls raise `IndexError`, but CLI arguments are already validated and no current repository caller supplies zero.
- Follow repository `AGENTS.md` and Conventional Commits when committing implementation work.
- This issue captures a plan for a new session. No background-removal implementation has been made yet.
